src/lib/utils/cpuid/cpuid_riscv64/cpuid_riscv64.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+* (C) 2025 Jack Lloyd
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/internal/cpuid.h>
+
+#include <botan/internal/os_utils.h>
+
+#if defined(BOTAN_TARGET_OS_IS_LINUX)
+   #include <asm/hwprobe.h>
+   #include <sys/hwprobe.h>
+
+   #define BOTAN_TARGET_HAS_RISCV_HWPROBE
+#endif
+
+namespace Botan {
+
+namespace {
+
+constexpr uint64_t bitflag(uint64_t b) {
+   return (static_cast<uint64_t>(1) << b);
+}
+
+template <typename... Bs>
+constexpr uint64_t bitflag(uint64_t b, Bs... rest) {
+   return bitflag(b) | bitflag(rest...);
+}
```

Comment:
```suggestion
template <std::convertible_to<uint64_t>... Bs>
constexpr uint64_t bitflag(Bs... bs) {
   return ((uint64_t(1) << bs) | ... | 0);
}
```

Fold expressions ftw. https://godbolt.org/z/T695nz7Wb

Reply:
On a second thought, `bitflag()` without parameters seems silly:

```C++
template <std::convertible_to<uint64_t>... Bs>
    requires(sizeof...(Bs) > 0)
constexpr uint64_t bitflag(Bs... bs) {
   return ((uint64_t(1) << bs) | ...);
}
```